### PR TITLE
adding SlacPgpCardG4 support

### PR DESCRIPTION
**Pull request by ruck314**

### CODE REVIEW

File: python/lcls2_pgp_fw_lib/hardware/SlacPgpCardG4/_Hsio.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env python3
```
Comment (bengineerd) [inline python/lcls2_pgp_fw_lib/hardware/SlacPgpCardG4/_Hsio.py]: This line in the header is only needed if the file is a script that will be executed directly from the command line.

**Comment by ruck314** [comment]
@bengineerd: SlacPgpCardG4/_SlacPgpCardG4Hsio.py now matches class name

**Comment by bengineerd** [comment]
Ok, I'll do the fixes for TimingRx.

File: hardware/SlacPgpCardG4/xdc/Pgp3Timing.xdc [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+create_generated_clock -name pgp3PhyTxClk0  [get_pins -hier -filter {name =~ */GEN_LANE[4].GEN_PGP3.U_Lane/REAL_PGP.U_Pgp/*/TXOUTCLK}]
+create_generated_clock -name pgp3PhyTxClk1  [get_pins -hier -filter {name =~ */GEN_LANE[5].GEN_PGP3.U_Lane/REAL_PGP.U_Pgp/*/TXOUTCLK}]
+create_generated_clock -name pgp3PhyTxClk2  [get_pins -hier -filter {name =~ */GEN_LANE[6].GEN_PGP3.U_Lane/REAL_PGP.U_Pgp/*/TXOUTCLK}]
+create_generated_clock -name pgp3PhyTxClk3  [get_pins -hier -filter {name =~ */GEN_LANE[7].GEN_PGP3.U_Lane/REAL_PGP.U_Pgp/*/TXOUTCLK}]
+
+######################
+# Timing Constraints #
+######################
+
+set_clock_groups -asynchronous \
```
Comment (bengineerd) [inline hardware/SlacPgpCardG4/xdc/Pgp3Timing.xdc]: Why do a separate `set_clock_groups` for each Phy clock here? Isn't it the same thing to just put all of these clocks in one big `set_clock_groups -asynchronous`?

```tcl
set_clock_groups -asynchronous \
    -group [get_clocks -include_generated_clocks {pgp3PhyRxClk0}] \
    -group [get_clocks -include_generated_clocks {pgp3PhyTxClk0}] \
    -group [get_clocks -include_generated_clocks {pgp3PhyRxClk1}] \
    -group [get_clocks -include_generated_clocks {pgp3PhyTxClk1}] \
    -group [get_clocks -include_generated_clocks {pgp3PhyRxClk2}] \
    -group [get_clocks -include_generated_clocks {pgp3PhyTxClk2}] \
    -group [get_clocks -include_generated_clocks {pgp3PhyRxClk3}] \
    -group [get_clocks -include_generated_clocks {pgp3PhyTxClk3}] \
    -group [get_clocks -include_generated_clocks {pgp3PhyRxClk4}] \
    -group [get_clocks -include_generated_clocks {pgp3PhyTxClk4}] \
    -group [get_clocks -include_generated_clocks {pgp3PhyRxClk5}] \
    -group [get_clocks -include_generated_clocks {pgp3PhyTxClk5}] \
    -group [get_clocks -include_generated_clocks {pgp3PhyRxClk6}] \
    -group [get_clocks -include_generated_clocks {pgp3PhyTxClk6}] \
    -group [get_clocks -include_generated_clocks {pgp3PhyRxClk7}] \
    -group [get_clocks -include_generated_clocks {pgp3PhyTxClk7}] \
    -group [get_clocks -include_generated_clocks {qsfpRefClkP}] \
    -group [get_clocks -include_generated_clocks {pciRefClkP}] \
    -group [get_clocks -include_generated_clocks {userClkP}]
```

Comment (ruck314) [reply hardware/SlacPgpCardG4/xdc/Pgp3Timing.xdc]: I think I tried that original but didn't work.

**Pull request merged**